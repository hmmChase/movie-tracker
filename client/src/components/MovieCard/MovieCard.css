.MovieCard {
  display: flex;
  flex-direction: column;
  width: 22%;
  height: 520px;
  border: 2px solid #000;
  border-radius: 8px;
  margin: 20px;
  text-align: left;
  background-color: grey;
  box-shadow: 0px 10px 15px 3px rgba(0, 0, 0, 0.55); }

.MovieCard:hover {
  box-shadow: 0px 5px 15px 3px rgba(118, 179, 181, 0.845); }

.favoriteToggle {
  position: relative;
  bottom: 55%;
  left: 84%;
  width: 50px;
  height: 50px;
  border: none;
  background-image: url(./assets/notFavorite.svg);
  background-repeat: no-repeat;
  background-size: 40px;
  color: transparent;
  font-size: 3rem;
  outline: none;
  background-color: transparent; }

.favoriteToggle:hover {
  cursor: pointer;
  transform: scale(1.3); }

.favorited .favoriteToggle {
  background-image: url(./assets/isFavorite.svg); }

.cardBody {
  padding: 0 20px;
  margin-top: -25px; }
  .cardBody h2 {
    margin-top: -0.8em;
    margin-bottom: .4em;
    font-size: 1.4em; }
  .cardBody p {
    margin-bottom: .4em;
    font-size: 1em; }
  .cardBody span {
    font-weight: bold; }

@media only screen and (max-width: 1200px) {
  .MovieCard {
    width: 20%; }
  .favoriteToggle {
    left: 80%; }
  .cardBody h2 {
    font-size: 1.2em; }
  .cardBody p {
    font-size: 0.8em; } }

@media only screen and (max-width: 1080px) {
  .MovieCard {
    width: 25%; } }

@media only screen and (max-width: 800px) {
  .MovieCard {
    width: 40%; } }

@media only screen and (max-width: 700px) {
  .MovieCard {
    width: 80%; } }
