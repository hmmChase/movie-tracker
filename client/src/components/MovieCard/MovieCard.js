import React from 'react';
import PropTypes from 'prop-types';
import './MovieCard.css';

const MovieCard = props => {
  const {
    movie_id,
    poster_path,
    title,
    release_date,
    vote_average,
    overview,
    toggleFavorite,
    isFavorite
  } = props;
  const baseUrl = 'https://image.tmdb.org/t/p/w200';
  const imageUrl = baseUrl + poster_path;
  const styleToggle = isFavorite(movie_id) ? 'favorited' : '';

  return (
    <div className={`MovieCard ${styleToggle}`} id={movie_id}>
      <img src={imageUrl} alt="movie poster" />
      <div className="favoriteToggle" onClick={() => toggleFavorite(movie_id)}>
        _
      </div>
      <div className="cardBody">
        <h2>{title}</h2>
        <p>
          <span className="bold-text">Release Date:</span> {release_date}
        </p>
        <p>
          <span className="bold-text">Vote Average:</span> {vote_average}
        </p>
        <p>
          {overview.length > 100
            ? overview.substring(0, 100) + '...'
            : overview}
        </p>
      </div>
    </div>
  );
};

MovieCard.propTypes = {
  movie_id: PropTypes.number.isRequired,
  poster_path: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  vote_average: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  overview: PropTypes.string.isRequired,
  toggleFavorite: PropTypes.func.isRequired,
  isFavorite: PropTypes.func.isRequired
};

export default MovieCard;
