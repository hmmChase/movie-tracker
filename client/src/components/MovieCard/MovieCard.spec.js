import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard';

describe('MovieCard', () => {
  let movieCard;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      movie_id: 12,
      poster_path: 'some_path',
      title: 'blitz',
      release_date: '12-13-14',
      vote_average: 8,
      overview: 'hi ho',
      toggleFavorite: jest.fn(),
      isFavorite: jest.fn()
    };
    movieCard = shallow(<MovieCard {...mockProps} />);
  });

  it('matches the snapshot', () => {
    expect(movieCard).toMatchSnapshot();
  });

  it('calls toggleFavorite when favoite button is clicked', () => {
    movieCard.find('.favoriteToggle').simulate('click');

    expect(mockProps.toggleFavorite).toHaveBeenCalledTimes(1);
  });
});
