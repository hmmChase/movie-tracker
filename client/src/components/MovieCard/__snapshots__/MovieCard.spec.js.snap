// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCard
    isFavorite={[Function]}
    movie_id={12}
    overview="hi ho"
    poster_path="some_path"
    release_date="12-13-14"
    title="blitz"
    toggleFavorite={[Function]}
    vote_average={8}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="movie poster"
          src="https://image.tmdb.org/t/p/w200some_path"
/>,
        <div
          className="favoriteToggle"
          onClick={[Function]}
>
          _
</div>,
        <div
          className="cardBody"
>
          <h2>
                    blitz
          </h2>
          <p>
                    <span
                              className="bold-text"
                    >
                              Release Date:
                    </span>
                     
                    12-13-14
          </p>
          <p>
                    <span
                              className="bold-text"
                    >
                              Vote Average:
                    </span>
                     
                    8
          </p>
          <p>
                    hi ho
          </p>
</div>,
      ],
      "className": "MovieCard ",
      "id": 12,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie poster",
          "src": "https://image.tmdb.org/t/p/w200some_path",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "_",
          "className": "favoriteToggle",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "_",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              blitz
</h2>,
            <p>
              <span
                            className="bold-text"
              >
                            Release Date:
              </span>
               
              12-13-14
</p>,
            <p>
              <span
                            className="bold-text"
              >
                            Vote Average:
              </span>
               
              8
</p>,
            <p>
              hi ho
</p>,
          ],
          "className": "cardBody",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "blitz",
            },
            "ref": null,
            "rendered": "blitz",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="bold-text"
>
                  Release Date:
</span>,
                " ",
                "12-13-14",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Release Date:",
                  "className": "bold-text",
                },
                "ref": null,
                "rendered": "Release Date:",
                "type": "span",
              },
              " ",
              "12-13-14",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="bold-text"
>
                  Vote Average:
</span>,
                " ",
                8,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Vote Average:",
                  "className": "bold-text",
                },
                "ref": null,
                "rendered": "Vote Average:",
                "type": "span",
              },
              " ",
              8,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "hi ho",
            },
            "ref": null,
            "rendered": "hi ho",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="movie poster"
            src="https://image.tmdb.org/t/p/w200some_path"
/>,
          <div
            className="favoriteToggle"
            onClick={[Function]}
>
            _
</div>,
          <div
            className="cardBody"
>
            <h2>
                        blitz
            </h2>
            <p>
                        <span
                                    className="bold-text"
                        >
                                    Release Date:
                        </span>
                         
                        12-13-14
            </p>
            <p>
                        <span
                                    className="bold-text"
                        >
                                    Vote Average:
                        </span>
                         
                        8
            </p>
            <p>
                        hi ho
            </p>
</div>,
        ],
        "className": "MovieCard ",
        "id": 12,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie poster",
            "src": "https://image.tmdb.org/t/p/w200some_path",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "_",
            "className": "favoriteToggle",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "_",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                blitz
</h2>,
              <p>
                <span
                                className="bold-text"
                >
                                Release Date:
                </span>
                 
                12-13-14
</p>,
              <p>
                <span
                                className="bold-text"
                >
                                Vote Average:
                </span>
                 
                8
</p>,
              <p>
                hi ho
</p>,
            ],
            "className": "cardBody",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "blitz",
              },
              "ref": null,
              "rendered": "blitz",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="bold-text"
>
                    Release Date:
</span>,
                  " ",
                  "12-13-14",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Release Date:",
                    "className": "bold-text",
                  },
                  "ref": null,
                  "rendered": "Release Date:",
                  "type": "span",
                },
                " ",
                "12-13-14",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="bold-text"
>
                    Vote Average:
</span>,
                  " ",
                  8,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Vote Average:",
                    "className": "bold-text",
                  },
                  "ref": null,
                  "rendered": "Vote Average:",
                  "type": "span",
                },
                " ",
                8,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "hi ho",
              },
              "ref": null,
              "rendered": "hi ho",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
